src/modules/RA/DataLayer/components/SourceFetcher.js

```diff
@@ -23,12 +23,13 @@ const SourceFetcher = ({ dispatch, dataProvider, sourceId, fields = [] }) => {
 
     const fillFields = async () => {
       const { data: { _type: type, fields: sourceFields = [] } } = await load(sourceId);
-      const filledFields = sourceFields.map(({ id, name, label }) => ({
-        id,
-        name,
-        label,
-        ...fields.find(({ id: fieldId }) => id === fieldId) || {},
-      }));
+      const filledFields = [
+        ...sourceFields.filter(({ id }) => !fields.find(({ id: fieldId }) => id === fieldId)),
```

Comment:
Can you un-inline this `filter` and the next `map` with an explicit var name to enhance readability ?